pkg/expressions/function/functions.go

```diff
@@ -150,22 +197,29 @@ func ChartFrom(
 ) expr.Option {
 	return expr.Function(
 		"chartFrom",
-		getChart(ctx, c, project, freightReqs, freightRefs),
+		getChartFromFreight(ctx, c, project, freightReqs, freightRefs),
 		new(func(repoURL string, chartName string, origin kargoapi.FreightOrigin) kargoapi.Chart),
 		new(func(repoURL string, chartName string) kargoapi.Chart),
 		new(func(repoURL string, origin kargoapi.FreightOrigin) kargoapi.Chart),
 		new(func(repoURL string) kargoapi.Chart),
 	)
 }
 
-// FreightMetadata returns an expr.Option that provides a `freightMetadata()` function for use in expressions.
+// ChartFromDiscoveredArtifacts returns an expr.Option that provides a
+// `chartFrom()` function for use, specifically, in expressions that define
+// criteria that permit or block automatic Freight creation after artifact
+// discovery.
 //
-// Usage:
-//   - `freightMetadata(freightRefName)` returns the entire metadata map for the Freight.
-//   - `freightMetadata(freightRefName, key)` returns the value for the given key
-//     (DEPRECATED; will be removed in v1.10).
-//
-// The second argument is deprecated as of v1.8. Prefer using the single-argument form.
+// The chartFrom function finds the latest Helm charts based on repository URL.
+func ChartFromDiscoveredArtifacts(artifacts *kargoapi.DiscoveredArtifacts) expr.Option {
+	return expr.Function(
+		"chartFrom",
+		getChartFromDiscoveredArtifacts(artifacts),
+		new(func(repoURL string) kargoapi.Chart),
+		new(func(repoURL string, chartName string) kargoapi.Chart),
```

Comment:
From previous experience with expr-lang, it's safest for longer, more complex signatures to precede shorter, less complex ones.

For reference: https://github.com/akuity/kargo/pull/2998#issue-2693283605

```suggestion
		new(func(repoURL string, chartName string) kargoapi.Chart),
		new(func(repoURL string) kargoapi.Chart),
```